plan/planbuilder.go

```diff
@@ -406,15 +406,17 @@ func (b *planBuilder) buildAnalyze(as *ast.AnalyzeTableStmt) LogicalPlan {
 	p := Analyze{}.init(b.allocator, b.ctx)
 	for _, tbl := range as.TableNames {
 		idxInfo, colInfo, pkInfo := getColsInfo(tbl)
-		result := Analyze{
-			TableInfo:   tbl.TableInfo,
-			IndicesInfo: idxInfo,
-			ColsInfo:    colInfo,
-			PkInfo:      pkInfo,
-		}.init(b.allocator, b.ctx)
-		result.SetSchema(expression.TableInfo2Schema(tbl.TableInfo))
-		addChild(p, result)
+		for _, idx := range idxInfo {
+			p.idxTasks = append(p.idxTasks, analyzeIndexTask{TableInfo: tbl.TableInfo, IndexInfo: idx})
```

Comment:
If there are multi-table, like `analyze table t1,t2`, we need to store it in the task.